cypress/e2e/spec.cy.ts

```diff
@@ -108,33 +116,140 @@ describe("testbox script", () => {
         postMessage(${fakeLoginMessageStringified});`);
     });
 
-    cy.get("@loginHandler").should("have.been.calledOnceWith", fakeLoginMessage().testbox.data);
+    cy.get("@loginHandlerSpy").should("have.been.calledOnceWith", fakeLoginMessage().testbox.data);
   });
 
-  it("Login event waiting for more than 10 seconds", () => {
+  it("Login handler should return redirect url", () => {
+    const fakeRedirectUrl = "https://fakeurl.com/success"
     cy.visit(BASE_URL, {
       onBeforeLoad: spyPostMessage,
-      onLoad: spyLoginHandler,
+      onLoad: (win) => {
+        loginHandlerStub(win, fakeRedirectUrl)
+        spyNavigateHandler(win)
+      },
     });
 
     cy.window().then((win) => {
-      win.eval(`test_startTestBox({...window.test_baseTbxConfig, window: window.parent });`);
+      win.eval(`test_startTestBox({...window.test_baseTbxConfig, window: window.parent, navigateHandler: window.test_fakeNavigateHandler });`);
+    });
+
+     cy.window().then((win) => {
+      win.eval("window.test_registerLoginHandler(window.test_fakeLoginHandler);");
+      win.eval(`
+        postMessage(${fakeLoginMessageStringified});`);
+    });
+
+    cy.get("@loginHandlerStub").should("have.returned", fakeRedirectUrl);
+    cy.get("@navigateHandlerSpy").should("have.been.calledWith", {url: fakeRedirectUrl});
+  });
+
+  it("LoginHandler should send login-failed message in case of error", () => {
+    cy.visit(BASE_URL, {
+      onBeforeLoad: spyPostMessage,
+      onLoad: (win) => loginHandlerStub(win, false),
     });
 
     cy.window().then((win) => {
+      win.eval(`test_startTestBox({...window.test_baseTbxConfig, window: window.parent });`);
+    });
+
+     cy.window().then((win) => {
+      win.eval("window.test_registerLoginHandler(window.test_fakeLoginHandler);");
       win.eval(`
         postMessage(${fakeLoginMessageStringified});`);
+    });
+
+    cy.get("@loginHandlerStub").should("have.returned", false);
+
+    cy.get("@postMessage").should("have.been.calledWith", {
+     testbox: {
+       version: 1,
+       event: "login-fail",
+       data: { message: undefined },
+       sender: "partner"
+      }
     })
+  });
+
+  it("LoginHandler should send login-success message in case of success", () => {
+    cy.visit(BASE_URL, {
+      onBeforeLoad: spyPostMessage,
+      onLoad: spyLoginHandler,
+    });
+
+    cy.window().then((win) => {
+      win.eval(`test_startTestBox(window.test_baseTbxConfig);`);
+    });
+
+    assertInitializeCall();
 
-    cy.wait(10500)
+    cy.window().then((win) => {
+      win.eval("window.test_registerLoginHandler(window.test_fakeLoginHandler);");
+      win.eval(`
+        postMessage(${fakeLoginMessageStringified});`);
+    });
 
+    cy.wrap(win => {
+      cy.get("@loginHandlerSpy").should("have.been.calledWith", true)
+    })
```

Comment:
Two things here
- I dont see any use for the `win` provided by the warp, which leads me to believe its not needed.
- This assertion seems incorrect, as the login handler should be called with the login event, and not a boolean.

Reply:
It should be `have.returned` instead of `have.been.calledWith`, not sure why test is passing, do you want me to remove this check or update it to `have.returned`?

Reply:
Same as the other comment, no point in checking what the mocked method is returning.